### PR TITLE
feat: allow to pass spec inline

### LINKED ISSUE #27: Inline spec
It would be nice if the `spec` would take a literal string containing what's usually in the file. This would allow for generating these, e.g. using `jsonnet`.

### PR DESCRIPTION
**Pull request by sh0rez**
This allows to pass the contents from `spec` directly from `drone.yml`, for use
with autogenerated drone configurations.

See https://github.com/drone-plugins/drone-manifest/issues/27#issuecomment-521269261 for details why this is required.

Fixes #27

### CODE REVIEW

File: plugin.go [hunk 1]
```diff
@@ -87,10 +87,16 @@ func (p *Plugin) Exec() error {
 	args = append(args, "push")
 
 	if p.Config.Spec != "" {
-		raw, err := ioutil.ReadFile(p.Config.Spec)
-
-		if err != nil {
-			return errors.Wrap(err, "failed to read template")
+		var raw []byte
+		// if spec is a filename, read it
+		if _, err := os.Stat(p.Config.Spec); os.IsNotExist(err) {
```
Comment (tboerger) [inline plugin.go]: Nope, you try to read a file that doesn't exist...

Comment (tboerger) [reply plugin.go]: https://golangr.com/file-exists/#error-checking

Comment (sh0rez) [reply plugin.go]: Alright, this one was misordered. Thanks!

**Comment by tboerger** [comment]
@drone-plugins/maintainers what do you think about that? IMHO it doesn't make a big difference if a previous step simply generates the manifest template which gets used by this plugin or if it's assigned as a multiline value.

I personally would prefer the previous step, showcased at https://github.com/drone-plugins/drone-manifest/issues/27#issuecomment-521271947.

**Comment by bradrydzewski** [comment]
I do not really have strong opinions.

I read the comment in issue #27 and I think the only question I have is how you plan on generating the string with jsonnet? Have you tested generating the string to ensure it works as expected before we modify the plugin accordingly?

In terms of implementation, I think if we allow reading from an inline string, we should use a different attribute name. I know in the past we have re-used the same field, however, I think being more explicit is better.

for example:

```yaml
spec: path/to/manifest.tmpl
spec_string: |
  image: ...
  manifests: ...
```

**Comment by bradrydzewski** [comment]
@tboerger this is more of an aside, but one thing I noticed with the manifest plugin is that we tend to re-use the same templates for most (all?) of our projects and we only change the image name. Is there a way to represent this in yaml (sort of like auto_tag) so we can make the template optional?

**Comment by tboerger** [comment]
Currently the template don't support the auto tagging, if we add that to the template it would be doable for most parts, but our windows versioning doesn't match the exact naming and I haven't checked how manifest is handling the arm versions without the template file.

**Comment by sh0rez** [comment]
What's the state of this? Can it be merged?

**Pull request merged**